src/services/heroService.js

```diff
@@ -54,6 +55,55 @@ class HeroService {
           error => reject(error));
     });
   }
+
+  retrieveHeroById(heroId) {
+    return new Promise((resolve, reject) => {
+      const query = `SELECT * FROM heroes WHERE id = ?;
+    SELECT name, id, type, active FROM equipment
+    WHERE heroId = ?;
+    SELECT equipmentId, attributeName, value FROM equipment
+    JOIN equipmentAttributes ON equipment.id = equipmentAttributes.equipmentId
+    JOIN attributeModifier ON attributeModifier.id = equipmentAttributes.attributeId
+    WHERE heroId = ?;`;
+      this.conn.query(query, [heroId, heroId, heroId], (err, rows) => {
+        if (err) {
+          reject(err);
+        } else if (rows[0].length > 0) {
+          const equipment = rows[1];
+          const modifiers = rows[2];
+          const hero = rows[0][0];
+          const inventory = [];
+
+          equipment.forEach((element) => {
+            const sortedModifiers = modifiers.filter(e => e.equipmentId === element.id);
+            sortedModifiers.forEach((e) => {
+              delete e.equipmentId;
+            });
+            const activeState = Boolean(Number(element.active));
+            const equipmentItem = new Equipment(element.name, sortedModifiers, element.type, activeState);
+            inventory.push(equipmentItem);
+          });
+
+          const resHero = new Hero(
+            hero.id,
+            hero.name,
+            hero.userId,
+            hero.experience,
+            hero.level,
+            hero.healthmax,
+            hero.healthact,
+            hero.attackmin,
+            hero.attackmax,
+            hero.defense,
+            inventory,
+            hero.finalWords);

```

Comment:
this instance creation will change with the modified constructor

Reply:
Sorted.